Jenkinsfile

```diff
@@ -82,45 +49,39 @@ pipeline {
   stages {
     stage('Update CI') {
       steps {
-        setGitHubBuildStatus('update-ci', 'Update Platform CI', 'PENDING')
         container('base') {
-          echo """
-          ----------------------------------------------------------------------
-          Synchronize K8s cluster state with Helmfile: Jenkins
-          Namespace: ${NAMESPACE}
-          ----------------------------------------------------------------------"""
-          echo "Override Helm 2 with Helm 3 since `--helm-binary /usr/bin/helm3` doesn't seem to work in `helmfile deps`"
-          sh 'mv /usr/bin/helm3 /usr/bin/helm'
-          echo 'Helm 3 version:'
-          sh 'helm version'
-
-          echo 'Helmfile version:'
-          sh 'helmfile version'
-
-          echo 'synchronize cluster state'
-          helmfileTemplate("${HELMFILE_ENVIRONMENT}", 'target')
-          withCredentials([
-            usernamePassword(credentialsId: 'packages.nuxeo.com-auth', usernameVariable: 'PACKAGES_USERNAME', passwordVariable: 'PACKAGES_PASSWORD'),
-            usernamePassword(credentialsId: 'connect-prod', usernameVariable: 'CONNECT_USERNAME', passwordVariable: 'CONNECT_PASSWORD'),
-          ]) {
+          nxWithGitHubStatus(context: 'update-ci', message: 'Update Platform CI') {
             script {
-              // not using usernamePassword since we need to fetch credentials from the target namespace, e.g. platform-staging if on a PR
-              def nexusUsername = getSecretData("${NEXUS_SECRET}", 'username')
-              def nexusPassword = getSecretData("${NEXUS_SECRET}", 'password')
-              def chartmuseumUsername = getSecretData("${CHARTMUSEUM_SECRET}", 'BASIC_AUTH_USER')
-              def chartmuseumPassword = getSecretData("${CHARTMUSEUM_SECRET}", 'BASIC_AUTH_PASS')
-              // not using usernamePassword to avoid displaying the access key id in the Jenkins credentials view
-              def awsAccessKeyId = getSecretData("${AWS_CREDENTIALS_SECRET}", 'access_key_id')
-              def awsSecretAccessKey = getSecretData("${AWS_CREDENTIALS_SECRET}", 'secret_access_key')
-              withEnv([
+              echo """
+              ----------------------------------------------------------------------
+              Synchronize K8s cluster state with Helmfile: Jenkins
+              Namespace: ${NAMESPACE}
+              ----------------------------------------------------------------------"""
+              echo 'Helmfile version:'
+              sh 'helmfile version'
+
+              echo 'synchronize cluster state'
+              nxHelmfile.template(namespace: env.NAMESPACE, file: 'helmfile.yaml', environment: env.HELMFILE_ENVIRONMENT, outputDir: 'target')
```

Comment:
Any reason to use `env.XXX` instead of `"$XXX"`?, just curious

Reply:
This is because in the `env.XXX` form there's fewer characters than in the `"${XXX}"` form.